File: src/routes/protected.js
import express from "express";
import { jwtAuth } from "../middleware/jwtAuth.js";
import { requireRole } from "../middleware/roleGuard.js";

const router = express.Router();

router.get("/me", jwtAuth, (req, res) => {
  res.json({ id: req.user.id, email: req.user.email, role: req.user.role });
});

router.get("/admin-only", jwtAuth, requireRole("admin"), (req, res) => {
  res.json({ secret: "only admins can see this" });
});

router.get("/user-only", jwtAuth, requireRole("user"), (req, res) => {
  res.json({ secret: "only users with role user can see this" });
});

export default router;

File: src/routes/protected.py
from flask import Blueprint, request, jsonify
from middleware.jwt_auth import jwt_auth
from middleware.role_guard import require_role

router = Blueprint("protected", __name__)


@router.get("/me")
@jwt_auth
def me():
    return {
        "id": request.user["id"],
        "email": request.user["email"],
        "role": request.user["role"]
    }


@router.get("/admin-only")
@jwt_auth
@require_role("admin")
def admin_only():
    return {"secret": "only admins can see this"}


@router.get("/user-only")
@jwt_auth
@require_role("user")
def user_only():
    return {"secret": "only users with role user can see this"}
